src/awkward/contents/recordarray.py

```diff
@@ -35,6 +35,15 @@
 numpy = Numpy.instance()
 
 
+def _apply_record_reducer(
+    reducer, layout: RecordArray, mask: bool, offsets: ak.index.Index, behavior
+) -> Content:
+    # Build a 1D list over these contents
+    array = wrap_layout(ak.contents.ListOffsetArray(offsets, layout), behavior=behavior)
+    # Perform the reduction
+    return ak.to_layout(reducer(array, mask))
```

Comment:
I don't yet know where to put this. It feels a bit odd creating a highlevel array _inside_ code written in a `Content` class definition, so I've moved it to a top-level function, much like the other reducer overloads are defined.